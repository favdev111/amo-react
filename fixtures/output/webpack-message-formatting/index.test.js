const {
  bootstrap,
  getOutputDevelopment,
  getOutputProduction,
} = require('../../utils');
const fs = require('fs-extra');
const path = require('path');
const Semaphore = require('async-sema');
const tempy = require('tempy');

describe('webpack message formatting', () => {
  const semaphore = new Semaphore(1, { capacity: Infinity });
  let testDirectory;
  beforeAll(async () => {
    testDirectory = tempy.directory();
    await bootstrap({ directory: testDirectory, template: __dirname });
  });
  beforeEach(async () => {
    await semaphore.acquire();
  });
  afterEach(async () => {
    fs.removeSync(path.join(testDirectory, 'src', 'App.js'));
    semaphore.release();
  });

  it('formats babel syntax error', async () => {
    fs.copySync(
      path.join(__dirname, 'src', 'AppBabel.js'),
      path.join(testDirectory, 'src', 'App.js')
    );

    const response = await getOutputProduction({ directory: testDirectory });
    expect(response).toMatchSnapshot();
  });

  it('formats css syntax error', async () => {
    fs.copySync(
      path.join(__dirname, 'src', 'AppCss.js'),
      path.join(testDirectory, 'src', 'App.js')
    );

    const response = await getOutputProduction({ directory: testDirectory });
    expect(response).toMatchSnapshot();
  });

  it('formats unknown export', async () => {
    fs.copySync(
      path.join(__dirname, 'src', 'AppUnknownExport.js'),
      path.join(testDirectory, 'src', 'App.js')
    );

    const response = await getOutputProduction({ directory: testDirectory });
    expect(response).toMatchSnapshot();
  });

  it('formats aliased unknown export', async () => {
    fs.copySync(
      path.join(__dirname, 'src', 'AppAliasUnknownExport.js'),
      path.join(testDirectory, 'src', 'App.js')
    );

    const response = await getOutputProduction({ directory: testDirectory });
    expect(response).toMatchSnapshot();
  });

  it('formats no default export', async () => {
    fs.copySync(
      path.join(__dirname, 'src', 'AppNoDefault.js'),
      path.join(testDirectory, 'src', 'App.js')
    );

    const response = await getOutputProduction({ directory: testDirectory });
    expect(response).toMatchSnapshot();
  });

  it('formats missing package', async () => {
    fs.copySync(
      path.join(__dirname, 'src', 'AppMissingPackage.js'),
      path.join(testDirectory, 'src', 'App.js')
    );

    const response = await getOutputProduction({ directory: testDirectory });
    expect(response).toMatchSnapshot();
  });

  it('formats eslint warning', async () => {
    fs.copySync(
      path.join(__dirname, 'src', 'AppLintWarning.js'),
      path.join(testDirectory, 'src', 'App.js')
    );

    const response = await getOutputProduction({ directory: testDirectory });
    const sizeIndex = response.stdout.indexOf('File sizes after gzip');
    if (sizeIndex !== -1) {
      response.stdout = response.stdout.substring(0, sizeIndex);
    }
    expect(response).toMatchSnapshot();
  });

  it('formats eslint error', async () => {
    fs.copySync(
      path.join(__dirname, 'src', 'AppLintError.js'),
      path.join(testDirectory, 'src', 'App.js')
    );

    const response = await getOutputProduction({ directory: testDirectory });
    expect(response).toMatchSnapshot();
  });

  it('helps when users tries to use sass', async () => {
    fs.copySync(
      path.join(__dirname, 'src', 'AppSass.js'),
      path.join(testDirectory, 'src', 'App.js')
    );

    const response = await getOutputProduction({ directory: testDirectory });
    expect(response).toMatchSnapshot();
  });

  it('formats file not found error', async () => {
    fs.copySync(
      path.join(__dirname, 'src', 'AppUnknownFile.js'),
      path.join(testDirectory, 'src', 'App.js')
    );

    const response = await getOutputProduction({ directory: testDirectory });
    expect(response).toMatchSnapshot();
  });

  it('formats case sensitive path error', async () => {
    fs.copySync(
      path.join(__dirname, 'src', 'AppIncorrectCase.js'),
      path.join(testDirectory, 'src', 'App.js')
    );

    const response = await getOutputDevelopment({ directory: testDirectory });
    if (process.platform === 'darwin') {
      expect(response.stderr).toMatch(
        `Cannot find file: 'export5.js' does not match the corresponding name on disk: './src/Export5.js'.`
      );
    } else {
      expect(response.stderr).not.toEqual(''); // TODO: figure out how we can test this on Linux/Windows
      // I believe getting this working requires we tap into enhanced-resolve
      // pipeline, which is debt we don't want to take on right now.
    }
  });

  it('formats out of scope error', async () => {
    fs.copySync(
      path.join(__dirname, 'src', 'AppOutOfScopeImport.js'),
      path.join(testDirectory, 'src', 'App.js')
    );

    const response = await getOutputProduction({ directory: testDirectory });
    expect(response).toMatchSnapshot();
  });
});
