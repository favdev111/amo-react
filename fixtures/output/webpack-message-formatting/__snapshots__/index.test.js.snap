// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack message formatting formats babel syntax error 1`] = `
Object {
  "stderr": "Creating an optimized production build...
Failed to compile.

./src/App.js
Syntax error: Unterminated JSX contents (8:12)

   6 |       <div>
   7 |         <span>
>  8 |       </div>
     |             ^
   9 |     );
  10 |   }
  11 | }


",
  "stdout": "",
}
`;

exports[`webpack message formatting formats css syntax error 1`] = `
Object {
  "stderr": "Creating an optimized production build...
[31mFailed to compile.
[39m
[7m./src/AppCss.css[27m
Syntax Error: (3:2) Unexpected }

 [90m 1 | [39m[33m.App[39m [33m{[39m
 [90m 2 | [39m  color[33m:[39m red[33m;[39m
[31m[1m>[22m[39m[90m 3 | [39m[33m}[39m[33m}[39m
 [90m   | [39m [31m[1m^[22m[39m
 [90m 4 | [39m


",
  "stdout": "",
}
`;

exports[`webpack message formatting formats eslint error 1`] = `
Object {
  "stderr": "Creating an optimized production build...
Failed to compile.

./src/App.js
  Line 4:  'b' is not defined  no-undef

Search for the keywords to learn more about each error.


",
  "stdout": "",
}
`;

exports[`webpack message formatting formats eslint warning 1`] = `
Object {
  "stderr": "",
  "stdout": "Creating an optimized production build...
Compiled with warnings.

./src/App.js
  Line 3:  'foo' is defined but never used  no-unused-vars

Search for the keywords to learn more about each warning.
To ignore, add // eslint-disable-next-line to the line before.

",
}
`;

exports[`webpack message formatting formats missing package 1`] = `
Object {
  "stderr": "Creating an optimized production build...
[31mFailed to compile.
[39m
Module not found: Error: Can't resolve 'unknown-package' in '/private/var/folders/c3/vytj6_h56b77f_g72smntm3m0000gn/T/bf26e1d3700ad14275f6eefb5e4417c1/src'


",
  "stdout": "",
}
`;

exports[`webpack message formatting formats unknown export 1`] = `
Object {
  "stderr": "Creating an optimized production build...
Failed to compile.

./src/App.js
1:1677-1680 './AppUnknownExport' does not contain an export named 'bar'.


",
  "stdout": "",
}
`;
